documentation/gcp-account-and-identity-architecture.md

```diff
@@ -0,0 +1,105 @@
+# GCP Account and Identity Architecture
+
+This document describes the Helpful Engineering account and identity architecture for Google Cloud Platform (GCP).
+
+## Problem Statement
+
+[Helpful Engineering](https://helpfulengineering.org) is creating and deploying applications to GCP as part of 
+its mission to address the COVID-19 pandemic. These applications are needed or developed by project teams 
+staffed by volunteers from all over the world.
+
+The applications and project teams develop and operate independently, have limited time, and use a wide set of 
+implementation technologies. Helpful Engineering has neither the time nor the staff to support high touch, high coordination
+solutions.
+
+## Solution Benefits
+
+Helpful Engineering needs an GCP account and identity architecture that maximizes data safety, team autonomy, and
+execution efficiency.
+
+The solution should:
+
+* Enable project delivery teams to deploy applications safely and independently.
+* Safeguard customer and organization data from external and internal threats, both accidental and malicious.
+* Minimize demand for scarce DevOps & Cloud skills.
+
+## Key Use Cases
+
+### UC1 - Deliver & Operate Applications Independently
+
+HelpfulEng project teams and deployments are independent, do not generally depend on each other, and expect a high 
+degree of autonomy. Project teams need to be able to develop applications, deliver them, and perform some amount of 
+operational activities in a self-service, low coordination manner.
+
+We anticipate that project teams will use a mix of manual and automated Cloud configuration and application deployment.
+Manual configurations are likely to be used in early prototyping and development. The DevOps team will enable and 
+encourage automated configurations delivery and operation to production. The DevOps team has no plans to enforce that.
+
+### UC2 - Provision Accounts
+
+The HelpfulEng DevOps team will provision GCP accounts for both shared and project delivery accounts. The DevOps team
+would like provision these accounts in a standardized way with low effort and simple adoption of Cloud security and
+governance practices.
+
+## Logical Architecture
+
+This design provisions a set of GCP accounts for each project team to deliver their applications and a few shared
+accounts for prototyping, security, and governance. The Helpful Engineering GCP Organization will look like: 
+
+```
+helpfulengineering.org Org
+└ Core
+  └ Log
+  └ Audit
+└ Sandboxes
+  └ helpful-gcp-sandbox2
+└ Project Delivery
+  └ Monitoring O2
+    └ helpful-project-monitoring-O2-dev
+    └ helpful-project-monitoring-O2-prod
+  └ <project name>
+    └ helpful-<project name>-dev
+    └ helpful-<project name>-prod
+  └ ...
```

Comment:
Annotating the resource hierarchy to show what types of resources are at different levels.  Not sure if this is accurate or helpful, so dispose of it if not.

```suggestion
helpfulengineering.org (organization resource)
└ Core  (folder resource)
  └ Log  (folder resource)
  └ Audit
└ Sandboxes  (folder resource)
  └ helpful-gcp-sandbox2  (project resource) 
└ Project Delivery  (folder resource)
  └ Monitoring O2  (folder resource)
    └ helpful-project-monitoring-O2-dev  (project resource)
    └ helpful-project-monitoring-O2-prod
  └ <project name> (folder resource)
    └ helpful-<project name>-dev  (project resource)
    └ helpful-<project name>-prod
  └ ...
```